akka-http-backend/src/main/scala/com/softwaremill/sttp/akkahttp/AkkaHttpBackend.scala

```diff
@@ -21,14 +21,48 @@ import scala.collection.immutable.Seq
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
 
-class AkkaHttpBackend private (
+trait AkkaHttpClient {
+  def singleRequest(
+      request: HttpRequest,
+      settings: ConnectionPoolSettings
+  ): Future[HttpResponse]
+}
+
+object AkkaHttpClient {
+  def fromAkkaHttpExt(
+      system: ActorSystem,
+      connectionContext: Option[HttpsConnectionContext],
+      customLog: Option[LoggingAdapter]
+  ): AkkaHttpClient = new AkkaHttpClient {
+
+    private val http = Http()(system)
+
+    override def singleRequest(
+        request: HttpRequest,
+        settings: ConnectionPoolSettings
+    ): Future[HttpResponse] = {
+      http.singleRequest(
+        request,
+        connectionContext.getOrElse(http.defaultClientHttpsContext),
+        settings,
+        customLog.getOrElse(system.log)
+      )
+    }
+  }
+
+  def fromStrict(run: HttpRequest => HttpResponse): AkkaHttpClient = new AkkaHttpClient {
+    def singleRequest(request: HttpRequest, settings: ConnectionPoolSettings): Future[HttpResponse] =
+      Future.successful(run(request))
```

Comment:
what about exceptions thrown by `run`? :)

Reply:
I assume it's pure... but of course we can loosen that assumption (or accept `HttpRequest => Try[HttpResponse]`).

In fact, I think the `fromStrict` function will be obsolete anyway, as the example I posted doesn't work for routes that don't immediately return a response (due to how `.response` works). `check` and a `Promise` will be required. I'll tell more when I'm able to run any of my tests with this :D